helix-core/src/main/java/org/apache/helix/model/Message.java

```diff
@@ -117,6 +120,11 @@
   // Currently, the field is only used for invalidating messages in controller's message cache.
   private boolean _expired = false;
 
+  // The expect period of time (in ms) that a message should be completed, default 1 day
+  public static final long MESSAGE_EXPECT_COMPLETION_PERIOD = HelixUtil
+      .getSystemPropertyAsLong(SystemPropertyKeys.MESSAGE_EXPECTED_COMPLETION_PERIOD,
+          24 * 60 * 60 * 1000 /* 1 day in ms */);
```

Comment:
This is legacy way of converting times. Better to use TimeUnit.DAYS.toMillis(1);

Reply:
Neat.